### ISSUE TITLE
Turn off MKL threading when used in parfor context.

### ISSUE THREAD

**Issue by DrTodd13** [numba/numba#2861]
If there is a MKL call in a parfor loop body, there can be an oversubscription problem.  Investigate whether we can insert some code to turn off MKL threading just around the parfor context.

**Comment by DrTodd13** [comment]
This was opened at the suggestion that this issue might be effecting logistic regression but I am unable to confirm a speedup when OMP_NUM_THREADS is set to 1.  In fact, I see a slowdown because the outer np.dot would then execute sequentially.  Without a use case where this is actually happening, I suggest we keep the issue open because it could potentially happen but it should be low priority.
@ehsantn @stuartarchibald

**Comment by dhirschfeld** [comment]
Maybe a reason to use the TBB by default for both numba and the mkl?

I was very interested in the talk given by Anton at Scipy 2016:
 * http://conference.scipy.org/proceedings/scipy2016/pdfs/anton_malakhov.pdf
 * https://software.intel.com/en-us/blogs/2016/04/04/unleash-parallel-performance-of-python-programs

**Comment by anton-malakhov** [comment]
Right, this is why TBB module for Python was created. We are working to bring it to other distributions as well. Try running that example using `python -m tbb` - assuming you use tbb and numba installed from intel channel.
